src/Lecture3.hs

```diff
@@ -181,8 +196,13 @@ together only different elements.
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
-
+appendDiff3 :: (Semigroup f, Eq f) => f -> f -> f -> f
+appendDiff3 x y z = combi (nub [x, y, z])
+  where 
+    combi [x1] = x1
+    combi [x1, x2] = x1 <> x2
+    combi [x1, x2, x3] = x1 <> x2 <> x3
```

Comment:
`combi` is currently a partial function from the compiler point of view. It's not great but it's acceptable.

You can use the standard function [`foldr1`](https://hackage.haskell.org/package/base-4.17.0.0/docs/GHC-List.html#v:foldr1) to implement `appendDiff3` in a single line 🌚